lib/assert.js

```diff
@@ -114,37 +113,46 @@ assert.fail = fail;
 //                             expected: expected });
 assert.AssertionError = AssertionError;
 
-function getBuffer(fd, assertLine) {
-  var lines = 0;
-  // Prevent blocking the event loop by limiting the maximum amount of
-  // data that may be read.
-  var maxReads = 64; // bytesPerRead * maxReads = 512 kb
-  var bytesRead = 0;
-  var startBuffer = 0; // Start reading from that char on
-  const bytesPerRead = 8192;
-  const buffers = [];
-  do {
-    const buffer = Buffer.allocUnsafe(bytesPerRead);
-    bytesRead = readSync(fd, buffer, 0, bytesPerRead);
-    for (var i = 0; i < bytesRead; i++) {
-      if (buffer[i] === 10) {
-        lines++;
-        if (lines === assertLine) {
-          startBuffer = i + 1;
-        // Read up to 15 more lines to make sure all code gets matched
-        } else if (lines === assertLine + 16) {
-          buffers.push(buffer.slice(startBuffer, i));
-          return buffers;
+function getBuffer(filename, assertLine) {
+  // Lazy load
+  const { openSync, closeSync, readSync } = require('fs');
```

Comment:
Perhaps we could do more traditional lazy loading and avoid the call to `require()` every time, by checking the `undefined`-ness of one of the variables.

Something like:

```js
{
  let openSync;
  let closeSync;
  let readSync;
  var getBuffer = function getBuffer(filename, assertLine) {
    if (openSync === undefined)
      ({ openSync, closeSync, readSync } = require('fs'));

    // ...
  };
}
```